provider/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinScriptHost.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.support
+
+import org.gradle.api.Action
+import org.gradle.api.internal.initialization.ClassLoaderScope
+import org.gradle.api.internal.plugins.DefaultObjectConfigurationAction
+import org.gradle.api.plugins.ObjectConfigurationAction
+
+import org.gradle.groovy.scripts.ScriptSource
+
+import org.gradle.internal.service.ServiceRegistry
+
+import org.gradle.kotlin.dsl.fileOperationsFor
+import org.gradle.kotlin.dsl.invoke
+
+
+class KotlinScriptHost(
```

Comment:
:x: This is now only used for init scripts and makes implementation for various targets different. Could this be used for other targets already? Doing this in a subsequent PR would be fine by me, but I'd like the implementations to be as similar as possible.

Reply:
Of course. But that's not related to supporting init scripts and it would change behaviour on the other targets so it is best handled as a separate issue.

Reply:
👍 Sounds good